trestle/tasks/cis_xlsx_to_oscal_catalog.py

```diff
@@ -0,0 +1,427 @@
+        # calculate output file name & check writability
+        oname = 'catalog.json'
+        ofile = opth / oname
+        if not overwrite and pathlib.Path(ofile).exists():
+            logger.warning(f'output: {ofile} already exists')
+            return TaskOutcome('failure')
+        xlsx_helper = XlsxHelper(ifile)
+        catalog_helper = CatalogHelper(title, version)
+        if xlsx_helper.is_ocp():
+            self._process_ocp(xlsx_helper, catalog_helper)
+        else:
+            self._process_rhel(xlsx_helper, catalog_helper)
+        catalog = catalog_helper.get_catalog()
+        # write OSCAL ComponentDefinition to file
+        if _verbose:
+            logger.info(f'output: {ofile}')
+        catalog.oscal_write(pathlib.Path(ofile))
+        return TaskOutcome('success')
+
+    def _process_ocp(self, xlsx_helper: XlsxHelper, catalog_helper: CatalogHelper) -> None:
+        """Process OCP."""
+        # transform each row into OSCAL equivalent
+        for row in xlsx_helper.row_generator():
+            section = xlsx_helper.get(row, 'section #')
+            recommendation = xlsx_helper.get(row, 'recommendation #')
+            title = xlsx_helper.get(row, 'title')
+            # init
+            props = []
+            parts = []
+            links = []
+            # props
+            self._add_property(xlsx_helper, props, row, 'profile')
+            self._add_property(xlsx_helper, props, row, 'status')
+            self._add_property(xlsx_helper, props, row, 'assessment status')
+            if recommendation is None:
+                frag = section
+            else:
+                frag = recommendation
+            # parts
+            self._add_part(xlsx_helper, parts, f'CIS-{frag}_smt', row, 'statement')
+            self._add_part(xlsx_helper, parts, f'CIS-{frag}_rat', row, 'rationale statement')
+            self._add_part(xlsx_helper, parts, f'CIS-{frag}_imp', row, 'impact statement')
+            self._add_part(xlsx_helper, parts, f'CIS-{frag}_rem', row, 'remediation procedure')
+            self._add_part(xlsx_helper, parts, f'CIS-{frag}_aud', row, 'audit procedure')
+            self._add_part(xlsx_helper, parts, f'CIS-{frag}_inf', row, 'additional information')
+            self._add_part(xlsx_helper, parts, f'CIS-{frag}_ctl', row, 'CIS Controls')
+            # group or control
+            if recommendation is None:
+                catalog_helper.add_group(section, title, props, parts)
+            else:
+                self._add_property_boolean(xlsx_helper, props, row, 'v7 IG1')
+                self._add_property_boolean(xlsx_helper, props, row, 'v7 IG2')
+                self._add_property_boolean(xlsx_helper, props, row, 'v7 IG3')
+                self._add_property_boolean(xlsx_helper, props, row, 'v8 IG1')
+                self._add_property_boolean(xlsx_helper, props, row, 'v8 IG2')
+                self._add_property_boolean(xlsx_helper, props, row, 'v8 IG3')
```

Comment:
Please remove duplicates

Reply:
These are not duplicates. Each function all has a unique set of parameters.